A backport of https://github.com/rust-lang/rust/pull/93350 from rust.
This was merged upstream as 1ca8d0bf8c154b0f602fadac8c38f14df5dea77a.

--- a/library/unwind/src/libunwind.rs
+++ b/library/unwind/src/libunwind.rs
@@ -246,6 +246,7 @@ if #[cfg(not(all(target_os = "ios", target_arch = "arm")))] {
     extern "C-unwind" {
         pub fn _Unwind_RaiseException(exception: *mut _Unwind_Exception) -> _Unwind_Reason_Code;
     }
+    #[link(name = "unwind", kind = "static", modifiers = "-bundle")]
     extern "C" {
         pub fn _Unwind_Backtrace(trace: _Unwind_Trace_Fn,
                                  trace_argument: *mut c_void)
